dietpi/dietpi-software

```diff
@@ -8296,7 +8303,77 @@ _EOF_
 			G_CONFIG_INJECT 'SNAPCLIENT_OPTS[[:blank:]=]' "SNAPCLIENT_OPTS=\"-h $snapcast_server_ip -p $snapcast_server_port\"" /etc/default/snapclient
 
 		fi
+		
+		software_id=198 # File Browser
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+
+			Banner_Installing
+
+			# ARMv6		
+			local arch='armv6'
+
+			# ARMv7
+			if (( $G_HW_ARCH == 2 )); then
+				arch='armv7'
+
+			# ARMv8
+			elif (( $G_HW_ARCH == 3 )); then
+				arch='arm64'
 
+			# x86_64
+			elif (( $G_HW_ARCH == 10 )); then
+				arch='amd64'
+
+			fi
+
+			local fallback_url="https://github.com/filebrowser/filebrowser/releases/download/v2.17.1/linux-$arch-filebrowser.tar.gz"
+			Download_Install "$(curl -sSfL 'https://api.github.com/repos/filebrowser/filebrowser/releases/latest' | mawk -F\" "/\"browser_download_url\": .*\/linux-$arch-filebrowser\.tar\.gz\"/{print \$4}")" ./filebrowser/
+
+			# Reinstall
+			[[ -d '/opt/filebrowser' ]] && G_EXEC rm -R /opt/filebrowser
+
+			# Install
+			G_EXEC cp -a filebrowser /opt/
+			G_EXEC rm -R filebrowser
+
+			# User
+			[[ -d '/mnt/dietpi_userdata/filebrowser' ]] || G_EXEC mkdir -p /mnt/dietpi_userdata/filebrowser
+			Create_User -G dietpi -d /mnt/dietpi_userdata/filebrowser filebrowser
+
+			# Config if not exist
+			if [[ ! -f '/mnt/dietpi_userdata/filebrowser/filebrowser.db' ]]
+			then
+				G_EXEC cd /mnt/dietpi_userdata/filebrowser/
+				G_EXEC /opt/filebrowser/filebrowser config init
+				G_EXEC /opt/filebrowser/filebrowser config set -a 0.0.0.0 -p 8085 -r /mnt
+				G_EXEC_DESC="Setting up File Browser login user 'dietpi'" G_EXEC /opt/filebrowser/filebrowser users add dietpi $GLOBAL_PW
+				#G_EXEC /opt/filebrowser/filebrowser config export filebrowser.yaml
+			fi
+
+			# Permissions
+			G_EXEC chown -R filebrowser:dietpi /mnt/dietpi_userdata/filebrowser /opt/filebrowser
+
+			# Service
+			cat << '_EOF_' > /etc/systemd/system/filebrowser.service
+[Unit]
+Description=File Browser (DietPi)
+Wants=network-online.target
+After=network-online.target dietpi-boot.service
+
+[Service]
+SyslogIdentifier=File Browser
+User=filebrowser
+Group=dietpi
```

Comment:
There is one problem when defining a group here: Supplementary group memberships are ignored, at least in some cases. I remember playing around with this and sometimes they still worked, sometimes not, and I wasn't able to find out why.

However, usually it is better to set "dietpi" as primary group. This means using `Create_User -g dietpi ...` above (lower case "g") and skipping `Group=` here. So filebrowser naturally creates new files and dirs as "dietpi" group and it is then reliably possible to do e.g. `usermod -aG www-data filebrowser` to add it to the webserver group as well. There is no "filebrowser" group then, but it doesn't have any use anyway.

Reply:
done. Group removed from service definition and set `dietpi` as primary group